crypto/rand/rand_unix.c

```diff
@@ -254,6 +254,13 @@ static ssize_t sysctl_random(char *buf, size_t buflen)
 #  endif
 
 #  if defined(OPENSSL_RAND_SEED_GETRANDOM)
+
+#   if defined(__linux) && !defined(__NR_getrandom)
```

Comment:
Should there be a kernel version check as well?
It's probably not critical since the system call will fail and other entropy sources will be tried.

Otherwise, this looks good.

Reply:
No, if the kernel supports the call, we win.
Of course it is possible that getrandom will have a short delay
(1-2 minutes on ARM systems, until ~80 bits of entropy are available).